File: app/src/main/java/com/cradletrial/cradlevhtapp/view/ui/intro/WelcomeFragment.java
package com.cradletrial.cradlevhtapp.view.ui.intro;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;

import com.cradletrial.cradlevhtapp.R;
import com.cradletrial.cradlevhtapp.view.ui.reading.PatientInfoFragment;

import java.util.ArrayList;
import java.util.List;

/**
 * Confirm privacy policy with user
 */
public class WelcomeFragment extends IntroBaseFragment {
    private View mView;
    private List<CheckBox> checkBoxes = new ArrayList<>();

    public WelcomeFragment() {
        // Required empty public constructor
        TAG = PatientInfoFragment.class.getName();
    }

    public static WelcomeFragment newInstance() {
        WelcomeFragment fragment = new WelcomeFragment();
        return fragment;
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_intro_welcome, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        mView = view;
    }

    @Override
    public void onMyBeingDisplayed() {
        // may not have created view yet.
        if (mView == null) {
            return;
        }
        hideKeyboard();

    }


    @Override
    public boolean onMyBeingHidden() {
        // may not have created view yet.
        if (mView == null) {
            return true;
        }

        return true;
    }


}

File: app/src/main/java/com/cradle/neptune/view/ui/intro/WelcomeFragment.kt
package com.cradle.neptune.view.ui.intro

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.cradle.neptune.R
import com.cradle.neptune.view.ui.reading.PatientInfoFragment

/**
 * Confirm privacy policy with user
 */
class WelcomeFragment : IntroBaseFragment() {
    private var mView: View? = null
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        // Inflate the layout for this fragment
        return inflater.inflate(R.layout.fragment_intro_welcome, container, false)
    }

    override fun onViewCreated(
        view: View,
        savedInstanceState: Bundle?
    ) {
        super.onViewCreated(view, savedInstanceState)
        mView = view
    }

    override fun onMyBeingDisplayed() {
        // may not have created view yet.
        if (mView == null) {
            return
        }
        hideKeyboard()
    }

    override fun onMyBeingHidden(): Boolean {
        // may not have created view yet.
        return true
    }

    companion object {
        fun newInstance(): WelcomeFragment {
            return WelcomeFragment()
        }
    }

    init {
        // Required empty public constructor
        TAG = PatientInfoFragment::class.java.name
    }
}
